### PR TITLE
@jacsonsouza/feature/instructors

### PR DESCRIPTION
**Pull request by jacsonsouza**
## PR Template

Issue: #**187972073**

### Description

This feature allows the administrator to manage instructors within the system. Admins can create, edit, delete, and view instructor records. Instructors themselves do not have an account within the system.

### Motivation and Context

- **Efficient Management:** Provides administrators with the ability to manage instructor records seamlessly.
- **Data Integrity:** Ensures that instructor information is accurately recorded and maintained.
- **System Consistency:** Aligns with other admin functionalities by providing full CRUD operations for instructors.

### How has this been tested?

1. Unit Tests:

    - Validated model attributes (e.g., presence, format, uniqueness constraints);
    - Tested model behaviors to ensure correct data processing and business logic.

2. Controller Tests:

    - Verified redirections and responses to admin requests;
    - Ensured successful record creation, updates, and deletions;
    - Checked appropriate error handling for invalid input scenarios.

3. System Tests:

    - Ensured that instructor management pages render correctly;
    - Tested CRUD operations via the UI to confirm expected behavior;
    - Simulated different user interactions, including edge cases, to verify robustness.

### Screenshots (if appropriate):

Screenshot from 2025-02-20 19-26-28

Screenshot from 2025-02-20 19-32-04

Screenshot from 2025-02-20 19-32-24

### Types of changes

- [ ] fix: A bug fix
- [x] feat: A new feature
- [ ] refactor: A code change that neither fixes a bug nor adds a feature
- [ ] enhancement: An enhancement makes something better
- [ ] test: Adding missing tests or correcting existing tests
- [ ] perf: A code change that improves performance
- [ ] docs: Documentation only changes
- [ ] style: Changes that do not affect the meaning of the code (e.g. prettier format)
- [ ] ci: Changes to our CI configuration files and scripts
- [ ] chore: Changes to the build process or auxiliary tools and libraries functionality to not work as expected

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---

Hours Required:

### CODE REVIEW

File: .idea/misc.xml
Comment (jacsonsouza) [inline .idea/misc.xml:1]: Remover

File: app/helpers/admin/instructors_helper.rb
Comment (jacsonsouza) [inline app/helpers/admin/instructors_helper.rb:1]: Remover

File: app/javascript/application.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 // Configure your import map in config/importmap.rb. Read more: https://github.com/rails/importmap-rails
-import "@hotwired/turbo-rails"
-import "controllers"
-import "preserve_scroll"
+import "@hotwired/turbo-rails";
+import "controllers";
+import "preserve_scroll";
+
```
Comment (jacsonsouza) [inline app/javascript/application.js:5]: Remover espaço

File: app/models/instructor.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Instructor < ApplicationRecord
+  validates :email,
+            presence: true,
+            uniqueness: true,
+            format: { with: /\A[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}\z/ }
```
Comment (jacsonsouza) [inline app/models/instructor.rb:5]: Talvez colocar em constantes:
 - Regex.phone.

File: app/views/admin/instructors/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                    url: admin_instructor_path,
+                    method: :patch, 
+                    html: { class: "space-y-4 md:space-y-6", data: { controller: "quill-form" } } do |f| %>
+
+   <%= f.input :email %>
+   <%= f.input :name %>
+   <%= f.input :phone, input_html: { data: { controller: "phone-mask", action: "input->phone-mask#apply" }} %>
+   <%= f.input :resume, as: :rich_text_area %>
+
+   <label class="inline-flex items-center cursor-pointer">
```
Comment (jacsonsouza) [inline app/views/admin/instructors/edit.html.erb:15]: Talvez extrair para um partial

File: config/importmap.rb [hunk 4]
```diff
@@ -5,4 +5,7 @@
 pin '@hotwired/stimulus', to: 'stimulus.min.js'
 pin '@hotwired/stimulus-loading', to: 'stimulus-loading.js'
 pin 'preserve_scroll', to: 'lib/preserve_scroll.js'
+
 pin_all_from 'app/javascript/controllers', under: 'controllers'
```
Comment (jacsonsouza) [inline config/importmap.rb:9]: Deixar na última linha

File: test/controllers/admin/instructors_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    assert_response :success
+  end
+
+  test 'should create an instructor' do
+    post admin_instructors_path,
+         params: { instructor: { email: 'test@test.com',
+                                 name: 'Instructor',
+                                 phone: '(42) 98425-2615',
+                                 resume: '{ ops: [{ insert: Faker::Lorem.paragraph(sentence_count: 2) + "\n" }] }' } }
```
Comment (jacsonsouza) [inline test/controllers/admin/instructors_controller_test.rb:20]: Corrigir

File: test/controllers/admin/instructors_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    assert_redirected_to admin_instructors_path
+    assert_equal t('flash_messages.created', model: Instructor.model_name.human), flash[:notice]
+  end
+
+  test 'should not create an instructor with invalid attributes' do
+    post admin_instructors_path,
+         params: { instructor: { email: '',
+                                 name: 'In',
+                                 phone: '(42) 98425-2615',
+                                 resume: '{ ops: [{ insert: Faker::Lorem.paragraph(sentence_count: 2) + "\n" }] }' } }
```
Comment (jacsonsouza) [inline test/controllers/admin/instructors_controller_test.rb:31]: Corrigir

File: test/controllers/uploads_controller_test.rb
Comment (jacsonsouza) [inline test/controllers/uploads_controller_test.rb:1]: Deletar

File: app/models/instructor.rb (hunk 2 above)
Comment (dmarczal) [reply app/models/instructor.rb:5]: Usar regex do Devise.

File: config/environments/development.rb [hunk 7]
```diff
@@ -82,4 +82,6 @@
   end
 
   config.web_console.allowed_ips = ['10.0.0.0/8', '172.16.0.0/12', '192.168.0.0/16']
+
+  config.active_storage.service = :cloudflare
```
Comment (dmarczal) [inline config/environments/development.rb:86]: Not save the local on cloudflare.

**Review by MarczalTSIGP — APPROVED** [review]
Review at PR https://github.com/MarczalTSIGP/maxiambiental/pull/17
